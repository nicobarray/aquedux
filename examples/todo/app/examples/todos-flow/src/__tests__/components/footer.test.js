// @flow

import React from 'react';
import configureStore from 'redux-mock-store';
import { shallow } from 'enzyme';

import Footer from '../../components/Footer';

const setup = (setupProps = {}) => {
  const store = configureStore()();
  const wrapper = shallow(<Footer store={store} />);

  return {
    store,
    wrapper
  };
};

describe('Footer', () => {
  test('renders without crashing', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });
});
