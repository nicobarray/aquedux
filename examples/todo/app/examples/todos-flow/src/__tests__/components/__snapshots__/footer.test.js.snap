// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders without crashing 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <p>
    Show:
     
    <Connect(Link)
        filter="SHOW_ALL"
    >
        All
    </Connect(Link)>
    , 
    <Connect(Link)
        filter="SHOW_ACTIVE"
    >
        Active
    </Connect(Link)>
    , 
    <Connect(Link)
        filter="SHOW_COMPLETED"
    >
        Completed
    </Connect(Link)>
</p>,
  "nodes": Array [
    <p>
      Show:
       
      <Connect(Link)
            filter="SHOW_ALL"
      >
            All
      </Connect(Link)>
      , 
      <Connect(Link)
            filter="SHOW_ACTIVE"
      >
            Active
      </Connect(Link)>
      , 
      <Connect(Link)
            filter="SHOW_COMPLETED"
      >
            Completed
      </Connect(Link)>
</p>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <Footer
        store={
                Object {
                        "clearActions": [Function],
                        "dispatch": [Function],
                        "getActions": [Function],
                        "getState": [Function],
                        "replaceReducer": [Function],
                        "subscribe": [Function],
                      }
        }
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "store": Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          },
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <p>
          Show:
           
          <Connect(Link)
                    filter="SHOW_ALL"
          >
                    All
          </Connect(Link)>
          , 
          <Connect(Link)
                    filter="SHOW_ACTIVE"
          >
                    Active
          </Connect(Link)>
          , 
          <Connect(Link)
                    filter="SHOW_COMPLETED"
          >
                    Completed
          </Connect(Link)>
</p>,
        "_debugID": 2,
        "_renderedOutput": <p>
          Show:
           
          <Connect(Link)
                    filter="SHOW_ALL"
          >
                    All
          </Connect(Link)>
          , 
          <Connect(Link)
                    filter="SHOW_ACTIVE"
          >
                    Active
          </Connect(Link)>
          , 
          <Connect(Link)
                    filter="SHOW_COMPLETED"
          >
                    Completed
          </Connect(Link)>
</p>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Footer
    store={
        Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          }
    }
/>,
}
`;
